src/Serialization/Address.purs

```diff
@@ -1,376 +1,283 @@
-module Serialization.Address
-  ( class Address
-  , AddressCsl
-  , BaseAddressCsl
-  , BaseAddress
-  , RewardAddress
-  , RewardAddressCsl
-  , PointerAddressCsl
-  , NetworkTag(TestnetTag, MainnetTag)
-  , StakeCred
-  , CredType
-  , StakeCredentialCsl
-  , PubKeyAddress
-  , ScriptAddress
-  , addressBech32
-  , addressBytes
-  , addrR
-  , baseAddressFromBech32
-  , baseAddressFromBytes
-  , rewardAddressFromBytes
-  , rewardAddressFromBech32
-  , ed25519KeyHashCredType
-  , baseAddress
-  , rewardAddress
-  , netTagToInt
-  , pubKeyAddress
-  , scriptAddress
-  , scriptHashCredType
-  , networkTag
-  , networkTagInt
-  , paymentCred
-  , delegationCred
-  , toAddressCsl
-  , intToNetTag
-  ) where
+module Serialization.Address where
 
 import Prelude
 
+import Control.Alt ((<|>))
+import Data.Function (on)
 import Data.Maybe (Maybe(..))
+import Data.Newtype (class Newtype)
+import Data.Typelevel.Undefined (undefined)
+import Data.UInt (UInt)
 import FfiHelpers (MaybeFfiHelper, maybeFfiHelper)
-import Serialization.Csl (class ToCsl, CslType, toCslRep, toCslType)
 import Serialization.Hash (Ed25519KeyHash, ScriptHash)
-import Types.Aliases (Bech32String)
+import Types.Aliases (Bech32String, Base58String)
 import Types.ByteArray (ByteArray)
 
--- | Denotes one of two possible values of Shelley address' network tag.
-data NetworkTag = TestnetTag | MainnetTag
+newtype Slot = Slot UInt
 
-derive instance Eq NetworkTag
+derive instance Eq Slot
+derive instance Newtype Slot _
+derive newtype instance Show Slot
 
-instance Show NetworkTag where
-  show = case _ of
-    TestnetTag -> "TestnetTag"
-    MainnetTag -> "MainnetTag"
+newtype TransactionIndex = TransactionIndex UInt
 
--- | Simple safety wrapper to ensure that only valid types are provided
--- | as address' stake credentials.
-newtype StakeCred a = StakeCred a
+derive instance Eq TransactionIndex
+derive instance Newtype TransactionIndex _
+derive newtype instance Show TransactionIndex
```

Comment:
I'd prefer not to use `newtype` instances for `Show`

Reply:
How it is worse than `genericShow`?

Reply:
Unless there is a compelling reason otherwise, I think a type's `Show` instance should produce a valid PS expression that would produce the shown value if evaluated